packages/lesswrong/server/vulcan-lib/apollo-server/initGraphQL.ts

```diff
@@ -153,6 +154,33 @@ const getGraphQLType = <N extends CollectionNameString>(
   }
 };
 
+/**
+ * Get the data needed to apply an access filter based on a graphql resolver
+ * return type.
+ */
+const getSqlResolverPermissionsData = (type: string|GraphQLScalarType) => {
+  // We only have access filters for return types that correspond to a collection.
+  if (typeof type !== "string") {
+    return null;
+  }
+
+  // We need to use a multi access filter for arrays, or a single access filter
+  // otherwise. We only apply the automatic filter for single dimensional arrays.
+  const isArray = type.indexOf("[") === 0 && type.lastIndexOf("[") === 0;
+
+  // Remove all "!"s (denoting nullability) and any array brackets to leave behind
+  // a type name string.
+  const nullableScalarType = type.replace(/[![\]]+/g, "");
```

Comment:
I'm so confused how this regex isn't unnecessarily escaping the `]` or missing the `[`

Reply:
![](https://imgs.xkcd.com/comics/perl_problems_2x.png)

Reply:
Rereading it it makes sense